### PR TITLE
Use dev and release builds of Status Desktop on the same machine

### LINKED ISSUE #5581: Create separate desktop versions for daily usage and dev/testing
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a developer, I want to use Status for both chatting and dev/testing so that I can always have a version of Status running for reliable chatting.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

To use Status for chatting everyday we need a reliable version for chatting whilst simultaneously running dev and test builds. Currently it is hard to use Status all the time when only one instance can be run at a time.

### Solution

Run a nightly build together with a dev/test build

### Additional Information

* Operating System: Desktop

As a reference it is similar to how you can run Chrome and Chrome Nightlies together.

screen shot 2018-08-22 at 8 29 17 am

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #5581 

This PR introduces two changes:
  - Use `REACT_SERVER_PORT` environment variable in both `ubuntu-server.js` and Qt. This addresses the problem of Node.js port conflicts in dev and release environments
  - Use different values for `ListenAddr` and `DataDir` when connecting to Status Node. This addresses the problem of `Geth is already running` when attempting to start a second instance of Status on the same machine. This requires a separate env variable `DEV_BUILD` that is set by default in `.env` file. Maybe there are better solutions?

For `ubuntu-server.js`, port value can be set either by (in descending precedence order):
1. Via `--port` command parameter, e.g. `node ./ubuntu-server.js --port=5001`
2. Via REACT_SERVER_PORT env var
3. Default value of 5000

In order to use a different port from release builds, pick a port value different from default (5000).

**TLDR**: when running in dev mode, set REACT_SERVER_PORT to a value different from default from 5000 and you should be good to go.

Relies on https://github.com/status-im/react-native-desktop/pull/351.

### CODE REVIEW

**Review by yenda — APPROVED** [review]

**Review by maxhora — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Comment by vkjr** [comment]
@siphiuel, if I understood correctly now we can concurrently run dev and prod version of Status because they have different ListenAddr and DataDir. And what about two concurrent prod versions launched by different users on the same machine? I believe that shouldn't work yet because ListenAddr is the same? Or I'm missing something?

**Comment by vitvly** [comment]
@vkjr Right, running 2 prod versions will not work. The value of 30303 (default Status node port) is hardcoded at https://github.com/status-im/status-go/blob/5b551c67fa51317e6dfdfbf4cfd25fe99c3beba8/params/config.go#L318. Machinery to support concurrent prod instances would require some status-go support I think.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-go">status-im/status-go</a></strong></div><div>The Status module that consumes go-ethereum. Contribute to status-im/status-go development by creating an account on GitHub.</div></blockquote>

**Review by rasom — APPROVED** [review]

**Comment by churik** [comment]
@siphiuel 
Steps:
1) change branch to `origin/feature/ubuntu-server-port`.
Added `export REACT_SERVER_PORT=5001` to `~/.bashrc` and source it.
2) in separate tab -> `npm install` -> `npm start`
3) in separate tab -> `node ./ubuntu-server.js`
4) in separate tab -> `lein figwheel-repl desktop `
5) in separate tab -> `react-native run-desktop`

When app is starting, got error "We were not able to decrypt your data" (there is no StatusIm folder in `Application Support`)
Logs are not created.

Am I doing something wrong?

`Apply` and `Cancel` don't work.

**Comment by vitvly** [comment]
@Churik it's an unrelated problem to this PR, the one falling under general `not able to decrypt`. Can you please try running on a clean repo? (after `git clean -fdx` which will remove Realm data under `status-react`)

**Comment by churik** [comment]
@siphiuel 
after running `git clean -fdx` and then `npm start` got an error:
```
npm start
npm ERR! missing script: start

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/curikovatm/.npm/_logs/2018-10-04T13_18_51_054Z-debug.log
```


Log:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@5.5.1
3 info using node@v8.11.2
4 verbose stack Error: missing script: start
4 verbose stack     at run (/usr/local/lib/node_modules/npm/lib/run-script.js:151:19)
4 verbose stack     at /usr/local/lib/node_modules/npm/lib/run-script.js:61:5
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:115:5
4 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:402:5
4 verbose stack     at handleExists (/usr/local/lib/node_modules/npm/node_modules/read-package-json/read-json.js:366:20)
4 verbose stack     at FSReqWrap.cb [as oncomplete] (fs.js:312:19)
5 verbose cwd /Users/curikovatm/status-desktop/react-native-desktop/react-native-cli/status-react
6 verbose Darwin 17.7.0
7 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
8 verbose node v8.11.2
9 verbose npm  v5.5.1
10 error missing script: start
11 verbose exit [ 1, true ]

```

**Comment by churik** [comment]
@siphiuel 
At attempt to start build from https://github.com/status-im/status-react/pull/6050#issuecomment-427272764 got an error:
Status cannot be opened because of a problem.
dev
Details:
```
ess:               Status [26001]
Path:                  /Applications/Status.app/Contents/MacOS/Status
Identifier:            im.status.statusim
Version:               0.1 (0.1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Status [26001]
User ID:               502

Date/Time:             2018-10-05 10:54:03.925 +0300
OS Version:            Mac OS X 10.13.6 (17G65)
Report Version:        12
Anonymous UUID:        93B7A170-78AE-1670-E0DF-EF45503C77C4

Sleep/Wake UUID:       E4EB38AB-BAE6-4091-B62A-5C849EA6BE6E

Time Awake Since Boot: 58000 seconds
Time Since Wake:       5000 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @rpath/QtQuick.framework/Versions/5/QtQuick
  Referenced from: /Applications/Status.app/Contents/MacOS/Status
  Reason: image not found

Binary Images:
       0x10aa2b000 -        0x10b975ff7 +im.status.statusim (0.1 - 0.1) <BFFFC014-4AC3-3F10-9C85-C698D1A1D698> /Applications/Status.app/Contents/MacOS/Status
       0x11988a000 -        0x1198d4acf  dyld (551.4) <8A72DE9C-A136-3506-AA02-4BA2B82DCAF3> /usr/lib/dyld

Model: MacBookPro11,5, BootROM MBP114.0184.B00, 4 processors, Intel Core i7, 2,5 GHz, 16 GB, SMC 2.30f2
Graphics: AMD Radeon R9 M370X, AMD Radeon R9 M370X, PCIe
Graphics: Intel Iris Pro, Intel Iris Pro, Built-In
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533642465238412D50422020
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D54343147533642465238412D50422020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x152), Broadcom BCM43xx 1.0 (7.77.37.31.1a9)
Bluetooth: Version 6.0.7f10, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD SM0512G, 500,28 GB
USB Device: USB 3.0 Bus
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 27.1
```
With nightly 05/10/2018 everuthing works fine.
Can you please take a look?

**Comment by churik** [comment]
Can see this problem also in https://github.com/status-im/status-react/pull/6128, so it is not PR-specific

**Comment by vkjr** [comment]
@siphiuel, I think it would be beneficial to add documentation page for developers on how to launch 2 builds simultaneously. To not lose instructions within this PR. Wdyt?

**Comment by churik** [comment]
@siphiuel  can you please rebase it to latest nightly to prevent crash desktop-app on start (was fixed)?

**Comment by vitvly** [comment]
@churik done

**Comment by lukaszfryc** [comment]
I had the same issue in this PR: https://github.com/status-im/status-react/pull/5888#issuecomment-427062494. cc @jakubgs

**Comment by jakubgs** [comment]
Looks like reverting to Qt `5.9.1` on macos hosts didn't fix it.

**Comment by jakubgs** [comment]
@MaxRis managed to figure out the issue, the fix is being tested here: https://github.com/status-im/status-react/pull/6168
